query/excluder.go

```diff
@@ -0,0 +1,94 @@
+package query
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+//Excluder allows us to support different methods of excluding in the future
```

Comment:
✔️

I think we can use this same interface to detect paths we've seen and apply regex to that to check for leading/trailing slashes (this is also naively implemented with a simple map and no regex right now).

Let me know if you'd like to work on that as well (should be in a different PR).